Dockerfile

```diff
@@ -2,27 +2,42 @@
 FROM golang:1.18 as builder
 
 WORKDIR /workspace
+
+# Install libzmq to generate certificate (build takes longer, slightly larger image)
+RUN apt-get update && apt-get install -y libsodium-dev libzmq3-dev libczmq-dev
+
 # Copy the Go Modules manifests
 COPY go.mod go.mod
 COPY go.sum go.sum
-# cache deps before building and copying source so that we don't need to re-download as much
-# and so that source changes don't invalidate our downloaded layer
 RUN go mod download
 
+# Copy the go source
+# This used to copy specific directories, now we
+# want to reproduce the local build environment
+# COPY . /workspace
+
 # Copy the go source
 COPY main.go main.go
 COPY api/ api/
 COPY controllers/ controllers/
 COPY pkg pkg/
+COPY hack hack
+COPY Makefile Makefile
 
-# Build
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o manager main.go
+# CGO_ENABLED=0 GOOS=linux GOARCH=amd64 
+# Build, ensuring we use the correct keygen
+# Note that the original build command did not work here, so updated to mimic the Makefile logic
+#CGO_ENABLED=0 GOOS=linux GOARCH=amd64 CGO_CFLAGS="-I/usr/include" CGO_LDFLAGS="-L/usr/lib -lstdc++ -lczmq -lzmq" go build -a -o manager main.go
+RUN make build-container && chmod +x ./manager
 
-# Use distroless as minimal base image to package the manager binary
-# Refer to https://github.com/GoogleContainerTools/distroless for more details
-FROM gcr.io/distroless/static:nonroot
+# We can't use distroless https://github.com/GoogleContainerTools/distroless
```

Comment:
you could theoretically only use apt in the builder and then copy the library files while still using distroless

Reply:
Hmm I could try that - I decided to use the simple ubuntu base here more out of fear that I'd miss something needed. Will put on my TODO to try.

Reply:
okay did some testing - see my comments in https://github.com/flux-framework/flux-operator/issues/153#issuecomment-1526233521. It does make a slightly smaller image, but I'm not convinced it's worth the maintainer churn / the potential of a single path changing and breaking the whole thing. I'll leave the issue and associated / PR branch open so others can come and offer comment. Thanks for the suggestion!